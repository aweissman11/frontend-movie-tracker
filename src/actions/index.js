export const getUserLoggedIn = (id, name) => {
  return {
    type: 'GET_USER_LOGGED_IN',
    user: { id, name }
  }
}

export const logUserOut = () => {
  return {
    type:  'LOG_USER_OUT'
  }
}

export const displayLogin = () => {
  return {
    type: 'DISPLAY_LOGIN'
  }
}

export const displaySignUp = () => {
  return {
    type: 'DISPLAY_SIGN_UP'
  }
}

export const setFavorites = (favorites = []) => {
  return {
    type: 'SET_FAVORITES',
    favorites
  }
}

export const updateFilters = (filters) => {
  return {
    type: 'UPDATE_FILTERS',
    filters
  }
}
export const updateSearchQuery = (searchQuery) => {
  return {
    type: 'UPDATE_QUERY',
    searchQuery
  }
}

export const setMovieList = (movies) => {
  return {
    type: 'SET_MOVIE_LIST',
    movies
  }
}

export const setHasErrored = (bool) => {
  return {
    type: 'SET_HAS_ERRORED',
    hasErrored: bool
  }
}

export const setIsOk = (bool) => {
  return {
    type: 'SET_IS_NOT_OK',
    notOk: bool
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}