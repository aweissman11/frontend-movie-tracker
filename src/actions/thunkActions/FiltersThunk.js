import * as Cleaners from '../../utilities/cleaners';
import fetchCall from '../../utilities/fetchCall';
import { setMovieList, setHasErrored, setIsOk, isLoading } from '../index';

export const getMovieList = (filters, query = '') => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const movies = await fetchCall(Cleaners.getFullUrl(filters, query));
      if (movies === 'failed') {
        dispatch(setIsOk(true));
      }
      dispatch(isLoading(false));
      dispatch(setMovieList(movies));
    } catch(e) {
      dispatch(setHasErrored(true));
    }
  }
}

