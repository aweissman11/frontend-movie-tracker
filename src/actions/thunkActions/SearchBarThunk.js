import * as Cleaners from '../../utilities/cleaners';
import fetchCall from '../../utilities/fetchCall';

export const getMovieList = (filterProperties, searchQuery) => {
  return async (dispatch) => {
    try {
      const movies = await fetchCall(Cleaners.getFullUrl(filterProperties, searchQuery));
      dispatch({
        type: 'GET_MOVIE_LIST',
        movies: movies
      })
    } catch(e) {
      dispatch({ type: 'ERROR'})
    }
  }
}

