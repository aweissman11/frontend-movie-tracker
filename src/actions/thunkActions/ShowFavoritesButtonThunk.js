import * as Cleaners from '../../utilities/cleaners';
import { setMovieList, setHasErrored, setIsOk, isLoading } from '../index';

export const getMovieList = (IDs) => {
  return async (dispatch) => {
    dispatch(isLoading(true))
    try {
      const results = await Cleaners.getFavoritesInfo(IDs);
      if (results === 'failed') {
        dispatch(setIsOk(true))
      }
      dispatch(isLoading(false))
      const movies = {
        results
      }
      await dispatch(setMovieList(movies))
    } catch(e) {
      dispatch(setHasErrored(true))
    }
  }
}

