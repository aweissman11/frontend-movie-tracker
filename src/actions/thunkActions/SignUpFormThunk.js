import { createNewUser } from '../../utilities/userDatabaseFetch';
import { isLoading, getUserLoggedIn, setHasErrored } from '../index';

export const submitNewUser = (name, email, password) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const newUserResponse = await createNewUser({ name, email, password });
      if (newUserResponse.error) {
        dispatch(setHasErrored(true));
      }
      dispatch(isLoading(false));
      await dispatch(getUserLoggedIn(newUserResponse.id, name));
    } catch(error) {
      dispatch(setHasErrored(true));
    }
  }
}