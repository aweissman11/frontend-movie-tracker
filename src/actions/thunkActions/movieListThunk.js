import * as Cleaners from '../../utilities/cleaners';
import { isLoading, setMovieList, setFavorites, setHasErrored, setIsOk } from '../index';
import fetchCall from '../../utilities/fetchCall';

export const getMovieList = () => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const movies = await fetchCall(Cleaners.movieListUrl);
      if (movies === 'failed') {
        dispatch(setIsOk(true));
      }
      dispatch(isLoading(false));
      dispatch(setMovieList(movies));
    } catch(e) {
      dispatch(setHasErrored(true));
    }
  }
}

export const updateFavorites = (id) => {
  return async (dispatch) => {
    try {
      const favorites = await Cleaners.getCurrentFavorites(id);
      dispatch(setFavorites(favorites));
    } catch(e) {
      dispatch(setHasErrored(true));
    }
  }
}