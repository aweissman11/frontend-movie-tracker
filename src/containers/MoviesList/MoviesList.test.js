import React from 'react';
import { shallow } from 'enzyme';
import { MoviesList, mapStateToProps, mapDispatchToProps } from './index'
import { getMovieList, updateFavorites } from '../../actions/thunkActions/movieListThunk'
import { getUserLoggedIn } from '../../actions/'

describe('MoviesList', () => {
  it('should match the snapshot MoviesList', () => {
    const setFetchedMovies = jest.fn()
    const setFavorites = jest.fn()
    const user = {
      name: 'Laura', 
      id: 2
    }
    const wrapper = shallow(<MoviesList user={user} setFavorites={setFavorites} setFetchedMovies={setFetchedMovies} movies={{}}/>)
    expect(wrapper).toMatchSnapshot()
    })

  describe('mapStateToProps', () => {
    it('should map the state to props', () => {
      const user = {
          name: 'Laura',
          id: 1
      }

      const movies = {
        title: "Back To The Future"
      }

      const favorites = {
        title: 'Boyhood'
      } 

      const expected = {user, movies, favorites}

      const mockState = {
        user: user,
        movies: movies, 
        favorites: favorites
      }
      const mappedProps = mapStateToProps(mockState)
      expect (mappedProps).toEqual(expected)

  })
})

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a getMovieList action when setFetchedMovies is called', () => {

      const movies = [ {title: 'Mean Girls'}, {title: 'Spiderman'} ]
      
      const mockDispatch = jest.fn()
      const actionToDispatch = getMovieList(movies)
      
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setFetchedMovies(movies)
      
      expect(mockDispatch).toHaveBeenCalled
    })
    it('should dispatch with a updateFavorites action when setFavorites is called', () => {
      const favorites = [ {title: 'Mean Girls'}, {title: 'Spiderman'} ]
      
      const movies = [ {title: 'Mean Girls'}, {title: 'Spiderman'} ]
      const mockDispatch = jest.fn()
      
      const actionToDispatch = updateFavorites(favorites)
      
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setFavorites(favorites)

      expect(mockDispatch).toHaveBeenCalled
    })
    it('should dispatch with getUserLoggedIn action when logIn is called', () => {

      const user = {name: 'Laura', id: 1}

      const mockDispatch = jest.fn()

      const actionToDispatch = getUserLoggedIn({user})

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.logIn({user})

      expect(mockDispatch).toHaveBeenCalled
    })
  })
})

