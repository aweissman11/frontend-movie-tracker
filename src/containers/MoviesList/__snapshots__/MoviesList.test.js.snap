// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should match the snapshot MoviesList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesList
    movies={Object {}}
    setFavorites={
      [MockFunction] {
        "calls": Array [
          Array [
            2,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setFetchedMovies={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    user={
      Object {
        "id": 2,
        "name": "Laura",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <section
            className="header-wrapper"
          >
            <section
              className="left-side-header-btns"
            >
              <Connect(LogButton) />
              <Connect(ShowFavoritesBtn) />
            </section>
            <Connect(Filters) />
            <Connect(SearchBar) />
          </section>
        </header>,
        <section
          className="movies-wrapper"
        >
          <div
            className="movies-stripe"
          />
          
        </section>,
        <footer>
          <div
            className="footer-wrap"
          >
            <section
              className="logo-footer-section"
            >
              <Logo />
              <h1>
                <span>
                  movie
                </span>
                Tracker
              </h1>
            </section>
            <section
              className="scroll-social-wrapper"
            >
              <i
                className="fab fa-facebook scroll-social"
              />
              <i
                className="fab fa-instagram scroll-social"
              />
              <i
                className="fab fa-twitter scroll-social"
              />
            </section>
            <h3
              className="footer-copyright"
            >
              © 2018 MovieTracker - All Rights Reserved
            </h3>
          </div>
        </footer>,
      ],
      "className": "movies-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="header-wrapper"
          >
            <section
              className="left-side-header-btns"
            >
              <Connect(LogButton) />
              <Connect(ShowFavoritesBtn) />
            </section>
            <Connect(Filters) />
            <Connect(SearchBar) />
          </section>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <section
                className="left-side-header-btns"
              >
                <Connect(LogButton) />
                <Connect(ShowFavoritesBtn) />
              </section>,
              <Connect(Filters) />,
              <Connect(SearchBar) />,
            ],
            "className": "header-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(LogButton) />,
                  <Connect(ShowFavoritesBtn) />,
                ],
                "className": "left-side-header-btns",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="movies-stripe"
            />,
            "",
          ],
          "className": "movies-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "movies-stripe",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "",
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="footer-wrap"
          >
            <section
              className="logo-footer-section"
            >
              <Logo />
              <h1>
                <span>
                  movie
                </span>
                Tracker
              </h1>
            </section>
            <section
              className="scroll-social-wrapper"
            >
              <i
                className="fab fa-facebook scroll-social"
              />
              <i
                className="fab fa-instagram scroll-social"
              />
              <i
                className="fab fa-twitter scroll-social"
              />
            </section>
            <h3
              className="footer-copyright"
            >
              © 2018 MovieTracker - All Rights Reserved
            </h3>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <section
                className="logo-footer-section"
              >
                <Logo />
                <h1>
                  <span>
                    movie
                  </span>
                  Tracker
                </h1>
              </section>,
              <section
                className="scroll-social-wrapper"
              >
                <i
                  className="fab fa-facebook scroll-social"
                />
                <i
                  className="fab fa-instagram scroll-social"
                />
                <i
                  className="fab fa-twitter scroll-social"
                />
              </section>,
              <h3
                className="footer-copyright"
              >
                © 2018 MovieTracker - All Rights Reserved
              </h3>,
            ],
            "className": "footer-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Logo />,
                  <h1>
                    <span>
                      movie
                    </span>
                    Tracker
                  </h1>,
                ],
                "className": "logo-footer-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        movie
                      </span>,
                      "Tracker",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "movie",
                      },
                      "ref": null,
                      "rendered": "movie",
                      "type": "span",
                    },
                    "Tracker",
                  ],
                  "type": "h1",
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fab fa-facebook scroll-social"
                  />,
                  <i
                    className="fab fa-instagram scroll-social"
                  />,
                  <i
                    className="fab fa-twitter scroll-social"
                  />,
                ],
                "className": "scroll-social-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fab fa-facebook scroll-social",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fab fa-instagram scroll-social",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fab fa-twitter scroll-social",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "© 2018 MovieTracker - All Rights Reserved",
                "className": "footer-copyright",
              },
              "ref": null,
              "rendered": "© 2018 MovieTracker - All Rights Reserved",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <section
              className="header-wrapper"
            >
              <section
                className="left-side-header-btns"
              >
                <Connect(LogButton) />
                <Connect(ShowFavoritesBtn) />
              </section>
              <Connect(Filters) />
              <Connect(SearchBar) />
            </section>
          </header>,
          <section
            className="movies-wrapper"
          >
            <div
              className="movies-stripe"
            />
            
          </section>,
          <footer>
            <div
              className="footer-wrap"
            >
              <section
                className="logo-footer-section"
              >
                <Logo />
                <h1>
                  <span>
                    movie
                  </span>
                  Tracker
                </h1>
              </section>
              <section
                className="scroll-social-wrapper"
              >
                <i
                  className="fab fa-facebook scroll-social"
                />
                <i
                  className="fab fa-instagram scroll-social"
                />
                <i
                  className="fab fa-twitter scroll-social"
                />
              </section>
              <h3
                className="footer-copyright"
              >
                © 2018 MovieTracker - All Rights Reserved
              </h3>
            </div>
          </footer>,
        ],
        "className": "movies-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="header-wrapper"
            >
              <section
                className="left-side-header-btns"
              >
                <Connect(LogButton) />
                <Connect(ShowFavoritesBtn) />
              </section>
              <Connect(Filters) />
              <Connect(SearchBar) />
            </section>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <section
                  className="left-side-header-btns"
                >
                  <Connect(LogButton) />
                  <Connect(ShowFavoritesBtn) />
                </section>,
                <Connect(Filters) />,
                <Connect(SearchBar) />,
              ],
              "className": "header-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Connect(LogButton) />,
                    <Connect(ShowFavoritesBtn) />,
                  ],
                  "className": "left-side-header-btns",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "section",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "section",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="movies-stripe"
              />,
              "",
            ],
            "className": "movies-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "movies-stripe",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "",
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="footer-wrap"
            >
              <section
                className="logo-footer-section"
              >
                <Logo />
                <h1>
                  <span>
                    movie
                  </span>
                  Tracker
                </h1>
              </section>
              <section
                className="scroll-social-wrapper"
              >
                <i
                  className="fab fa-facebook scroll-social"
                />
                <i
                  className="fab fa-instagram scroll-social"
                />
                <i
                  className="fab fa-twitter scroll-social"
                />
              </section>
              <h3
                className="footer-copyright"
              >
                © 2018 MovieTracker - All Rights Reserved
              </h3>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <section
                  className="logo-footer-section"
                >
                  <Logo />
                  <h1>
                    <span>
                      movie
                    </span>
                    Tracker
                  </h1>
                </section>,
                <section
                  className="scroll-social-wrapper"
                >
                  <i
                    className="fab fa-facebook scroll-social"
                  />
                  <i
                    className="fab fa-instagram scroll-social"
                  />
                  <i
                    className="fab fa-twitter scroll-social"
                  />
                </section>,
                <h3
                  className="footer-copyright"
                >
                  © 2018 MovieTracker - All Rights Reserved
                </h3>,
              ],
              "className": "footer-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Logo />,
                    <h1>
                      <span>
                        movie
                      </span>
                      Tracker
                    </h1>,
                  ],
                  "className": "logo-footer-section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span>
                          movie
                        </span>,
                        "Tracker",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "movie",
                        },
                        "ref": null,
                        "rendered": "movie",
                        "type": "span",
                      },
                      "Tracker",
                    ],
                    "type": "h1",
                  },
                ],
                "type": "section",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fab fa-facebook scroll-social"
                    />,
                    <i
                      className="fab fa-instagram scroll-social"
                    />,
                    <i
                      className="fab fa-twitter scroll-social"
                    />,
                  ],
                  "className": "scroll-social-wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fab fa-facebook scroll-social",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fab fa-instagram scroll-social",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fab fa-twitter scroll-social",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "section",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "© 2018 MovieTracker - All Rights Reserved",
                  "className": "footer-copyright",
                },
                "ref": null,
                "rendered": "© 2018 MovieTracker - All Rights Reserved",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
